wcfsetup/install/files/lib/data/smiley/Smiley.class.php

```diff
@@ -75,26 +77,54 @@ public function getAliases()
     }
 
     /**
-     * Returns the height of the smiley.
+     * Returns the dimensions of the smiley.
      *
-     * @return  int
+     * @since   5.4
+     * @return  int[]
      */
-    public function getHeight()
+    public function getDimensions()
     {
-        if ($this->height === null) {
-            $this->height = 0;
+        if ($this->height === null && $this->width === null) {
+            $this->height = $this->width = 0;
 
             $file = WCF_DIR . $this->smileyPath;
             if (\file_exists($file) && \preg_match('~\.(gif|jpe?g|png)$~', $file)) {
                 $data = \getimagesize($file);
                 if ($data !== false) {
-                    // index '1' contains the height of the image
-                    $this->height = $data[1];
+                    [$this->width, $this->height] = $data;
                 }
             }
         }
 
-        return $this->height;
+        return [
+            'width' => $this->width,
+            'height' => $this->height,
+        ];
+    }
+
+    /**
+     * Returns the height of the smiley.
+     *
+     * @return  int
+     */
+    public function getHeight()
+    {
+        $dimensions = $this->getDimensions();
+
+        return $dimensions['height'];
```

Comment:
```suggestion
        return $this->getDimensions()['height'];
```
Increased readability, because there is no processing happening and the lines are already very short.